src/components/EditForm.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  ChangeEvent,
+  FC,
+  KeyboardEvent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+interface Props {
+  editFormSubmitHandler: (arg: string) => void,
+  setIsEditing: React.Dispatch<React.SetStateAction<boolean>>,
+  title: string
+}
+
+export const EditForm:FC<Props> = (
+  {
+    editFormSubmitHandler,
+    setIsEditing,
+    title,
+  },
+) => {
+  const [query, setQuery] = useState(title);
+  const textInput = useRef<HTMLInputElement | null>(null);
+
+  const cancelChange = () => {
+    setIsEditing(false);
+    setQuery(title);
+  };
+
+  const onEditTodoFormChange = (
+    event: ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+  };
+
+  const escKeyUpHandler = (
+    event: KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Escape') {
+      cancelChange();
+    }
+  };
+
+  const onFormSubmit = () => {
+    if (query !== title) {
+      editFormSubmitHandler(query.trim());
```

Comment:
It is good that there is a reduction of extra spaces here (`.trim()`), but it is better to transfer it to a separate variable and use it also in the `if` condition.